pkg/auth/oauth/handlers/default_auth_handler.go

```diff
@@ -113,6 +115,21 @@ func (authHandler *unionAuthenticationHandler) AuthenticationNeeded(apiClient au
 				w.WriteHeader(http.StatusFound)
 			default:
 				w.WriteHeader(http.StatusUnauthorized)
+				ctx, ok := authHandler.requestContextMapper.Get(req)
+				if !ok {
+					return false, fmt.Errorf("no context found for request to audit")
+				}
+				ev := request.AuditEventFrom(ctx)
+				if ev != nil {
```

Comment:
I assume we do not care about the `ev == nil` case?

Reply:
Nope, we don't. This means audit is not happening and there's nothing we need to do here.